### PR TITLE
Spglib functions added

### PR DESCRIPTION
**Pull request by psavery**
Added several spglib functions for crystals

Added space group perception, primitive reduction, and cell conventionalization.
In addition, the user can specify the tolerance for all of these features.
All of the features use spglib to perform them. When performing any action,
the program tells the user what the tolerance is currently set to.

### CODE REVIEW

**Comment by psavery** [comment]
This pull request is dependent upon this one: https://github.com/OpenChemistry/avogadrolibs/pull/73

The first commit is basically just that other pull request, so I wouldn't worry about that one here. It's the second commit that's important for this pull request. It has the added functions for primitive reduction and cell conventionalization. It also has the GUI features added. You can try it out on the files here: https://github.com/OpenChemistry/avogadrodata/pull/4

Basically, you can test it by performing a primitive reduction on the conventional corundum, and you can perform a conventionalization of primitive corundum. That's what I'm doing in the tests. It's making sure that the number of atoms and the volume changes like it should.

This also adds the user interface for perceiving a space group. You can test that on the corundum example or on the rutile.POSCAR example in avogadrodata/data now.

**Comment by ghutchis** [comment]
Ignore the Travis CI bit.. I'm working to get that going.

**Comment by psavery** [comment]
This has been updated to remove the merge conflict.
